src/main/java/hudson/plugins/matrix_configuration_parameter/MatrixCombinationsParameterValue.java

```diff
@@ -38,49 +45,45 @@
 
     @DataBoundConstructor
     public MatrixCombinationsParameterValue(String name, Boolean[] values, String[] confs) {
-        super(name, null);
-        this.values = values;
-        this.confs = confs;
+        this(name, values, confs, null);
     }
 
     public MatrixCombinationsParameterValue(String name, Boolean[] values, String[] confs, String description) {
         super(name,  description);
-        this.values = values;
-        this.confs = confs;
+        this.values = (values != null) ? values.clone() : null;
+        this.confs = (confs != null) ? confs.clone() : null;
     }
 
     public Boolean[] getValues() {
-        return values;
+        return (values != null) ? values.clone() : null;
     }
 
     public String[] getConfs() {
-        return confs;
+        return (confs != null) ? confs.clone() : null;
     }
 
     @Override
     public VariableResolver<String> createVariableResolver(AbstractBuild<?, ?> build) {
         return new VariableResolver<String>() {
             public String resolve(String name) {
+                if (!MatrixCombinationsParameterValue.this.name.equals(name)) {
+                    return null;
+                }
 
-                String parameterValue = "";
-
+                List<String> conds = new ArrayList<String>();
 
                 for (int uniqueIdIndex= 0 ; uniqueIdIndex < values.length ; uniqueIdIndex++){
                     Boolean value=values[uniqueIdIndex];
                     String conf = confs[uniqueIdIndex];
 
                     if (value.booleanValue()){
-                        parameterValue += "("+conf.replace("="," == '").replace(",","' && ")+"') || ";
+                        conds.add("("+conf.replace("="," == '").replace(",","' && ")+"')");
```

Comment:
<BugInstance type='SBSC_USE_STRINGBUFFER_CONCATENATION' priority='Normal' category='PERFORMANCE' message='hudson.plugins.matrix_configuration_parameter.MatrixCombinationsParameterValue$1.resolve(String) concatenates strings using + in a loop' lineNumber='73'/>